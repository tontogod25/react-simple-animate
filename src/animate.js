// @flow
import React from 'react';
import propsGenerator from './utils/propsGenerator';
import filterMountOrUnmount from './utils/filterMountOrUnmount';
import mapChildren from './utils/mapChildren';
import setDelayState from './utils/setDelayState';

export const defaultState = {
  willEnd: false,
  willStart: false,
  willComplete: false,
  willEnter: false,
  willLeave: false,
  played: false,
  childrenStoreInState: [],
};

export type Style = { [string]: string | number };

type ChildrenType = Array<React$Element<any>> | React$Element<any> | null;

export type Props = {
  startAnimation: boolean,
  children?: ChildrenType,
  startStyle?: Style,
  endStyle: Style,
  onCompleteStyle?: Style,
  durationSeconds?: number,
  delaySeconds?: number,
  reverseDelaySeconds?: number,
  easeType?: string,
  forceUpdate?: boolean,
  tag?: string,
  onComplete?: () => mixed,
  onError?: (Object, Object) => mixed,
  className?: string,
  animateOnAddRemove: boolean,
  render: Object => any,
  transition?: string,
  refCallback?: (React$Element<any>) => {},
};

export type State = {
  willEnd: boolean,
  willStart: boolean,
  willComplete: boolean,
  willEnter: boolean,
  willLeave: boolean,
  played: boolean,
  childrenStoreInState?: ChildrenType,
};

const timeDelayForChildEnterOrLeave = 0.005;

export default class Animate extends React.Component<Props, State> {
  static displayName = 'ReactSimpleAnimate';

  static defaultProps = {
    startStyle: {},
    onCompleteStyle: undefined,
    durationSeconds: 0.3,
    delaySeconds: 0,
    children: null,
    reverseDelaySeconds: 0,
    easeType: 'linear',
    forceUpdate: false,
    tag: 'div',
    onComplete: undefined,
    onError: undefined,
    className: undefined,
    transition: '',
    refCallback: undefined,
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      ...defaultState,
      ...(props.children ? { childrenStoreInState: props.children } : null),
    };
  }

  componentDidMount() {
    this.setDelayAndOnComplete(this.props);
  }

  componentWillReceiveProps(nextProps: Props) {
    const { startAnimation, children } = nextProps;
    const toggledAnimation = startAnimation !== this.props.startAnimation;

    this.setState({
      ...(toggledAnimation ? { ...{ ...defaultState } } : null),
      childrenStoreInState: children,
      played: toggledAnimation,
    });
  }

  shouldComponentUpdate(
    { startStyle, endStyle, startAnimation, children, forceUpdate }: Props,
    { willEnd, willStart, willComplete, willLeave, willEnter }: State,
  ) {
    // only situation that should trigger a re-render
    return (
      // object compare can be enahnced, but try to keep it simple here
      JSON.stringify(startStyle) !== JSON.stringify(this.props.startStyle) ||
      JSON.stringify(endStyle) !== JSON.stringify(this.props.endStyle) ||
      startAnimation !== this.props.startAnimation ||
      children !== this.props.children ||
      willEnd !== this.state.willEnd ||
      willStart !== this.state.willStart ||
      willComplete !== this.state.willComplete ||
      willLeave !== this.state.willLeave ||
      willEnter !== this.state.willEnter ||
      !!forceUpdate
    );
  }

  componentDidUpdate(previousProps: Props) {
    const { reverseDelaySeconds, animateOnAddRemove, startAnimation } = this.props;
    const toggledAnimation = startAnimation !== previousProps.startAnimation;
    // animation on mount or unmount
    if (animateOnAddRemove) this.setChildrenState(this.props);

    this.setDelayAndOnComplete(this.props, toggledAnimation && !startAnimation && !!reverseDelaySeconds);
  }

  componentDidCatch(error: Object, info: Object) {
    const { onError = false } = this.props;
    if (onError) onError(error, info);
  }

  componentWillUnmount() {
    this.delayTimeout && clearTimeout(this.delayTimeout);
    this.completeTimeout && clearTimeout(this.completeTimeout);
    this.leaveTimeout && clearTimeout(this.leaveTimeout);
    this.enterTimeout && clearTimeout(this.enterTimeout);
    this.delayTimeout = undefined;
    this.completeTimeout = undefined;
    this.leaveTimeout = undefined;
    this.enterTimeout = undefined;
  }

  setDelayAndOnComplete(
    { delaySeconds, startAnimation, onCompleteStyle, durationSeconds, onComplete, reverseDelaySeconds }: Props,
    isReverseWithDelay: boolean = false,
  ): void {
    // delay animation
    this.delayTimeout && clearTimeout(this.delayTimeout);

    if (delaySeconds && startAnimation) {
      this.delayTimeout = setDelayState.call(this, delaySeconds, 'willEnd');
    } else if (isReverseWithDelay) {
      // reverse animation
      this.delayTimeout = setDelayState.call(this, reverseDelaySeconds, 'willStart');
    }

    if ((!onComplete && !onCompleteStyle) || !startAnimation) return;

    this.completeTimeout && clearTimeout(this.completeTimeout);
    this.completeTimeout = setDelayState.call(
      this,
      parseFloat(delaySeconds) || parseFloat(durationSeconds) || 0,
      'willComplete',
      onComplete,
    );
  }

  setCurrentChildrenToState = () => {
    this.setState({
      childrenStoreInState: this.props.children,
    });
  };

  setChildrenState(nextProps: Props): void {
    const { childrenStoreInState } = this.state;
    const { children, startAnimation, durationSeconds } = nextProps;

    if (!Array.isArray(childrenStoreInState) || !Array.isArray(children) || !startAnimation) {
      return;
    }

    if (childrenStoreInState.length !== children.length) {
      const { mappedChildren, childrenWillUnmount, childrenWillMount } = filterMountOrUnmount(
        childrenStoreInState,
        children,
      );

      if (childrenWillUnmount && startAnimation) {
        this.leaveTimeout && clearTimeout(this.leaveTimeout);
        this.leaveTimeout = setDelayState.call(this, durationSeconds, 'willLeave', this.setCurrentChildrenToState);
      }

      this.setState({
        willEnter: false,
        willLeave: false,
        childrenStoreInState: mappedChildren,
      });

      if (childrenWillMount && startAnimation) {
        this.enterTimeout && clearTimeout(this.enterTimeout);
        this.enterTimeout = setDelayState.call(this, timeDelayForChildEnterOrLeave, 'willEnter', this.setCurrentChildrenToState);
      }
    } else if (!startAnimation) {
      this.enterTimeout && clearTimeout(this.enterTimeout);
      this.enterTimeout = setDelayState.call(this, timeDelayForChildEnterOrLeave, 'willEnter', this.setCurrentChildrenToState);

      this.setState({
        childrenStoreInState,
      });
    }
  }

  delayTimeout = null;
  completeTimeout = null;
  leaveTimeout = null;
  enterTimeout = null;

  render() {
    const { tag, children, animateOnAddRemove, render } = this.props;
    const tagName = tag || 'div';
    const componentProps = propsGenerator(this.props, this.state);
    const haveChildToAnimate = Array.isArray(children) && animateOnAddRemove;

    if (render && !haveChildToAnimate) {
      return render(componentProps);
    }
    return React.createElement(
      tagName,
      componentProps,
      haveChildToAnimate ? mapChildren(this.props, this.state) : children,
    );
  }
}
