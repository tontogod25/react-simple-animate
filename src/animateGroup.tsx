import * as React from 'react';
import { Sequences, AnimationProps, AnimateKeyframesProps } from './types';
import calculateTotalDuration from './utils/calculateTotalDuration';
import getSequenceId from './utils/getSequenceId';
import isUndefined from './utils/isUndefined';

const { useState, useRef, useEffect } = React;

export interface Props {
  play: boolean;
  sequences?: Sequences;
  children?: any;
}

export const AnimateContext = React.createContext({
  animationStates: {},
  register: (data: AnimationProps | AnimateKeyframesProps): void => {},
});

export default function AnimateGroup(props: Props) {
  const { play, sequences = [], children } = props;
  const [animationStates, setAnimationStates] = useState({});
  const animationsRef = useRef<{
    [key: string]: AnimationProps | AnimateKeyframesProps;
  }>({});

  const register = (data: AnimationProps | AnimateKeyframesProps) => {
    const { sequenceIndex, sequenceId } = data;
    if (isUndefined(sequenceId) && isUndefined(sequenceIndex)) return;

    animationsRef.current[getSequenceId(sequenceIndex, sequenceId)] = data;
  };

  useEffect((): void => {
    const sequencesToAnimate =
      Array.isArray(sequences) && sequences.length
        ? sequences
        : Object.values(animationsRef.current);
    const localAnimationState = {};

    (play ? sequencesToAnimate : [...sequencesToAnimate].reverse()).reduce(
      (previous, current, currentIndex) => {
        const { sequenceId, sequenceIndex, ...restAttributes } = current;
        const { duration: defaultDuration, delay, overlay } = restAttributes;
        const id = getSequenceId(sequenceIndex, sequenceId, currentIndex);
        const duration = defaultDuration || 0.3;
        const currentTotalDuration = calculateTotalDuration({
          duration,
          delay,
          overlay,
        });

        const totalDuration = currentTotalDuration + previous;

        localAnimationState[id] = {
          play,
          pause: !play,
          delay:
            currentIndex === 0
              ? delay || 0
              : delay
              ? currentTotalDuration
              : previous,
          controlled: true,
        };

        return totalDuration;
      },
      0,
    );

    setAnimationStates(localAnimationState);
  }, [play]);

  return (
    <AnimateContext.Provider value={{ animationStates, register }}>
      {children}
    </AnimateContext.Provider>
  );
}
