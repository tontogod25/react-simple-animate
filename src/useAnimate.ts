import * as React from 'react';
import msToSec from './utils/secToMs';
import { AnimationProps, Style } from './types';

const { useState, useEffect, useRef } = React;

export default function useAnimate(
  props: AnimationProps,
): {
  isPlaying: boolean;
  style: Style;
  play: (boolean) => void;
} {
  const {
    start,
    end,
    complete,
    onComplete,
    delay = 0,
    duration = 0.3,
    easeType = 'linear',
  } = props;
  const transition = `all ${duration}s ${easeType} ${delay}s`;
  const [style, setStyle] = useState<Style>({ ...start, transition });
  const [isPlaying, setIsPlaying] = useState(false);
  const onCompleteTimeRef = useRef<number>(0);

  useEffect(
    (): any => (): void => {
      onCompleteTimeRef.current && clearTimeout(onCompleteTimeRef.current);
    },
    [],
  );

  const play = (isPlay: boolean) => {
    setStyle({
      ...(isPlay ? end : start),
      transition,
    });

    setIsPlaying(isPlay);

    if (isPlay && (complete || onComplete)) {
      // @ts-ignore
      onCompleteTimeRef.current = setTimeout((): void => {
        complete && setStyle(complete);
        onComplete && onComplete();
      }, msToSec(delay + duration));
    }
  };

  return {
    isPlaying,
    style,
    play,
  };
}
