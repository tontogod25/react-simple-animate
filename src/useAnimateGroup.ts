import * as React from 'react';
import createRandomName from './utils/createRandomName';
import calculateTotalDuration from './utils/calculateTotalDuration';
import createTag from './logic/createTag';
import deleteRules from './logic/deleteRules';
import { HookSequences, Style } from './types';

const { useEffect, useState, useRef } = React;

interface Props {
  sequences: HookSequences;
}

function createArrayWithNumbers(length: number): null[] {
  return Array.from({ length }, (): null => null);
}

export default function useAnimateGroup(
  props: Props,
): { styles: (Style | null)[]; play: (boolean) => void; isPlaying: boolean } {
  const { sequences = [] } = props;
  const defaultArray = createArrayWithNumbers(sequences.length).map(
    (_, index) => props.sequences[index].start,
  ) as (Style)[];
  const [styles, setStyles] = useState(defaultArray);
  const [isPlaying, setPlaying] = useState(false);
  const animationNamesRef = useRef<
    { forward: string; reverse: string; animationName }[]
  >([]);
  const styleTagRef = useRef<{ forward?: string; reverse?: string }[]>([]);
  const playRef = useRef<(isPlay: boolean) => void>();

  useEffect((): any => {
    sequences.forEach(({ keyframes = false }, i): void => {
      if (!Array.isArray(keyframes)) return;

      if (!animationNamesRef.current[i]) {
        animationNamesRef.current[i] = {} as any;
        styleTagRef.current[i] = {};
      }

      animationNamesRef.current[i].forward = createRandomName();
      let result = createTag({
        animationName: animationNamesRef.current[i].forward,
        keyframes,
      });
      styleTagRef.current[i].forward = result.styleTag;

      animationNamesRef.current[i].reverse = createRandomName();
      result = createTag({
        animationName: animationNamesRef.current[i].reverse,
        keyframes: keyframes.reverse(),
      });
      styleTagRef.current[i].reverse = result.styleTag;
    });

    return (): void => {
      Object.values(animationNamesRef).forEach(
        ({ forward, reverse }, i): void => {
          if (!styleTagRef[i]) return;
          deleteRules(styleTagRef[i].sheet, forward);
          deleteRules(styleTagRef[i].sheet, reverse);
        },
      );
    };
  }, []);

  playRef.current = playRef.current
    ? playRef.current
    : (isPlay: boolean): void => {
        let totalDuration = 0;
        const animationRefWithOrder = isPlay
          ? animationNamesRef.current
          : [...animationNamesRef.current].reverse();
        const styles = (isPlay ? sequences : [...sequences].reverse()).map(
          (current, currentIndex): Style => {
            const {
              duration = 0.3,
              delay = 0,
              overlay,
              keyframes,
              iterationCount = 1,
              easeType = 'linear',
              direction = 'normal',
              fillMode = 'none',
              end = {},
              start = {},
            } = current;
            const delayDuration = currentIndex === 0 ? delay : totalDuration;
            const transition = `all ${duration}s ${easeType} ${delayDuration}s`;
            totalDuration =
              calculateTotalDuration({ duration, delay, overlay }) +
              totalDuration;

            return keyframes
              ? {
                  animation: `${duration}s ${easeType} ${delayDuration}s ${iterationCount} ${direction} ${fillMode} running ${
                    isPlay
                      ? animationRefWithOrder[currentIndex].forward
                      : animationRefWithOrder[currentIndex].reverse
                  }`,
                }
              : {
                  ...(isPlay ? end : start),
                  transition,
                };
          },
        );

        // @ts-ignore
        setStyles(isPlay ? styles : [...styles].reverse());
        setPlaying(!isPlaying);
      };

  return { styles, play: playRef.current, isPlaying };
}
