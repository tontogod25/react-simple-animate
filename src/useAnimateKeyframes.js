// @flow
import { useState, useEffect, useRef } from 'react';
import createRandomName from './utils/createRandomName';
import createTag from './style/createTag';
import type { AnimateKeyframesProps } from './animateKeyframes';
import deleteRules from './style/deleteRules';

type UseAnimateKeyframesProps = AnimateKeyframesProps & {
  animationName: string,
};

export default function useAnimateKeyframes(props: UseAnimateKeyframesProps) {
  const {
    durationSeconds = 0.3,
    delaySeconds = 0,
    easeType = 'linear',
    direction = 'normal',
    fillMode = 'none',
    iterationCount = 1,
    playState = 'running',
    keyframes,
  } = props;
  const animationNameRef = useRef('');
  const styleTagRef = useRef(null);
  const [{ play }, setPlay] = useState(props);
  const playFunction = (playValue: boolean) => {
    setPlay({
      ...props,
      play: playValue,
    });
  };

  useEffect(
    () => {
      if (!animationNameRef.current) {
        const name = createRandomName();
        const { styleTag } = createTag({ animationName: name, keyframes });
        styleTagRef.current = styleTag;
        animationNameRef.current = name;
      }

      return () => {
        if (styleTagRef.current && animationNameRef.current) {
          // $FlowIgnoreLine
          deleteRules(styleTagRef.current.sheet, animationNameRef.current);
        }
      };
    },
    [play],
  );

  const style = play
    ? {
        animation: `${durationSeconds}s ${easeType} ${delaySeconds}s ${iterationCount} ${direction} ${fillMode} ${playState} ${animationNameRef.current ||
          ''}`,
      }
    : null;

  return [
    {
      style,
      play,
    },
    (playValue: boolean) => playFunction(playValue),
  ];
}
