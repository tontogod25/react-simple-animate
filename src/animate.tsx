import * as React from 'react';
import { AnimateContext } from './animateGroup';
import secToMs from './utils/secToMs';
import getSequenceId from './utils/getSequenceId';
import isUndefined from './utils/isUndefined';
import { AnimationProps } from './types';

const { useEffect, useState, useRef, useContext } = React;

export default function Animate(props: AnimationProps) {
  const {
    play,
    children,
    render,
    start,
    end,
    complete = '',
    onComplete,
    delay = 0,
    duration = 0.3,
    easeType = 'linear',
    sequenceId,
    sequenceIndex,
  } = props;
  const onCompleteTimeRef = useRef<any>();
  const [style, setStyle] = useState(start || {});
  const { register, animationStates = {} } = useContext(AnimateContext);
  const id = getSequenceId(sequenceIndex, sequenceId);

  useEffect(() => {
    if ((!isUndefined(sequenceIndex) && sequenceIndex >= 0) || sequenceId)
      register(props);
  }, []);

  useEffect(() => {
    const animationState = animationStates[id] || {};

    setStyle({
      ...(play || animationState.play ? end : start),
      transition: `all ${duration}s ${easeType} ${parseFloat(
        animationState.delay || delay,
      )}s`,
    });

    if (play && (complete || onComplete)) {
      onCompleteTimeRef.current = setTimeout(() => {
        complete && setStyle(complete);
        onComplete && onComplete();
      }, secToMs(parseFloat(animationState.delay || delay) + duration));
    }

    return () => {
      onCompleteTimeRef.current && clearTimeout(onCompleteTimeRef.current);
    };
  }, [
    id,
    animationStates,
    play,
    duration,
    easeType,
    delay,
    onComplete,
    start,
    end,
    complete,
  ]);

  return render ? render({ style }) : <div style={style}>{children}</div>;
}
