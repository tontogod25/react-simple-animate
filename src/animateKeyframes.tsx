import * as React from 'react';
import createTag from './logic/createTag';
import createRandomName from './utils/createRandomName';
import deleteRule from './logic/deleteRules';
import { AnimateContext } from './animateGroup';
import getSequenceId from './utils/getSequenceId';
import { AnimateKeyframesProps } from './types';
import getPlayState from './utils/getPlayState';

const { useRef, useEffect, useContext, useState } = React;

export default function AnimateKeyframes(props: AnimateKeyframesProps) {
  const {
    children,
    play,
    render,
    duration = 0.3,
    delay = 0,
    easeType = 'linear',
    direction = 'normal',
    fillMode = 'none',
    iterationCount = 1,
    sequenceIndex,
    keyframes,
    sequenceId,
  } = props;
  const animationNameRef = useRef('');
  const id = getSequenceId(sequenceIndex, sequenceId);
  const styleTagRef = useRef({
    sheet: {},
  });
  const { register, animationStates = {} } = useContext(AnimateContext);
  const forceUpdate = useState(false)[1];

  useEffect(() => {
    animationNameRef.current = createRandomName();
    const { styleTag } = createTag({
      animationName: animationNameRef.current,
      keyframes,
    });
    styleTagRef.current = styleTag;
    register(props);

    if (play) forceUpdate(true);
    return (): void => deleteRule(styleTagRef.current.sheet, animationNameRef.current);
  }, []);

  const style = {
    animation: `${duration}s ${easeType} ${delay}s ${iterationCount} ${direction} ${fillMode} ${getPlayState(
      (animationStates[id] || {}).play || play,
    )} ${animationNameRef.current || ''}`,
  };

  return render ? render({ style }) : <div style={style || {}}>{children}</div>;
}
