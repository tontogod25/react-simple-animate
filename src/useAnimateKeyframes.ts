import { useState, useEffect, useRef, useContext } from 'react';
import createRandomName from './utils/createRandomName';
import createTag from './logic/createTag';
import { AnimateContext } from './animateGroup';
import deleteRules from './logic/deleteRules';
import { AnimateKeyframesProps, Style } from './types';
import getPlayState from './utils/getPauseState';

export default function useAnimateKeyframes(
  props: AnimateKeyframesProps,
): {
  style: Style;
  play: (boolean) => void;
  pause: (boolean) => void;
  isPlaying: boolean;
} {
  const {
    duration = 0.3,
    delay = 0,
    easeType = 'linear',
    direction = 'normal',
    fillMode = 'none',
    iterationCount = 1,
    keyframes,
  } = props;
  const animationNameRef = useRef({
    forward: '',
    reverse: '',
  });
  const styleTagRef = useRef({
    forward: { sheet: {} },
    reverse: { sheet: {} },
  });
  const { register } = useContext(AnimateContext);
  const [isPlaying, setIsPlaying] = useState(true);
  const [isPaused, setIsPaused] = useState(false);
  const playRef = useRef<(isPlay: boolean) => void>();

  useEffect((): any => {
    animationNameRef.current.forward = createRandomName();
    let result = createTag({
      animationName: animationNameRef.current.forward,
      keyframes,
    });
    styleTagRef.current.forward = result.styleTag;

    animationNameRef.current.reverse = createRandomName();
    result = createTag({
      animationName: animationNameRef.current.reverse,
      keyframes: keyframes.reverse(),
    });
    styleTagRef.current.reverse = result.styleTag;
    register(props);

    return (): void => {
      deleteRules(
        styleTagRef.current.forward.sheet,
        animationNameRef.current.forward,
      );
      deleteRules(
        styleTagRef.current.reverse.sheet,
        animationNameRef.current.reverse,
      );
    };
  }, []);

  playRef.current = playRef.current ? playRef.current: (isPlay: boolean): void => {
    setIsPlaying(isPlay);
  };

  const pause = (isPaused: boolean): void => {
    setIsPaused(isPaused);
  };

  const style = {
    animation: `${duration}s ${easeType} ${delay}s ${iterationCount} ${direction} ${fillMode} ${getPlayState(
      isPaused,
    )} ${(isPlaying
      ? animationNameRef.current.forward
      : animationNameRef.current.reverse) || ''}`,
  };

  return {
    style,
    play: playRef.current,
    pause,
    isPlaying,
  };
}
