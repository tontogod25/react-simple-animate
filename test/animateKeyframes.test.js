import renderer from 'react-test-renderer';
import React from 'react';
import { AnimateKeyframesChild } from '../src/animateKeyframes';

jest.mock('../src/style/createTag', () => () => 'createTag');
jest.mock('../src/utils/createRandomName', () => () => 'RSI-4dtcsgvdy');

describe('AnimateKeyframesChild', () => {
  const props = {
    keyframes: [],
    play: false,
    animationStates: [],
  };

  it('should render correctly', () => {
    const tree = renderer.create(<AnimateKeyframesChild {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when play set true', () => {
    const tree = renderer.create(<AnimateKeyframesChild {...{ ...props, play: true }} />);
    expect(tree).toMatchSnapshot();
  });
});
