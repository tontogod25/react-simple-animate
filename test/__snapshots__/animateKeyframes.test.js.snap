// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimateKeyframesChild should render correctly 1`] = `<div />`;

exports[`AnimateKeyframesChild should render correctly when play set true 1`] = `
<div
  style={
    Object {
      "animation": "0.3s linear 0s 1 normal none running RSI-4dtcsgvdy",
    }
  }
/>
`;

exports[`AnimateKeyframesChild should render correctly when render pro 1`] = `<div />`;

exports[`AnimateKeyframesChild should render correctly when render props is supplied 1`] = `<div />`;

exports[`AnimateKeyframesChild should render correctly with 1`] = `<div />`;

exports[`AnimateKeyframesChild should render correctly with AnimateKeyframesChild 1`] = `<div />`;

exports[`AnimateKeyframesChild should render correctly with AnimateKeyframesChild 2`] = `<div />`;
